### ISSUE TITLE
Built-in Substitution Live Preview Option or Plugin

### ISSUE THREAD

**Issue by pbnj-dragon** [vim/vim#14868]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**

When performing more complex substitutions in Vim (e.g. , `:s`, `:%s` , `:'<,'>s`), it is difficult to preview the outcome before executing the command. 

Being able to preview substitutions live before executing the command can help identify issues & typos (e.g. `&` vs `\&`) in regex patterns early and allow users to fix them before executing the command.

**Describe the solution you'd like**

A built-in Vim option or a built-in Vim plugin to enable substitution live previews.

**Describe alternatives you've considered**

- Neovim users have [`'inccommand'`](https://neovim.io/doc/user/options.html#'inccommand') option.
- Vim users have [traces.vim](https://github.com/markonm/traces.vim) plugin.

**Additional context**

N/A

**Comment by nv-quan** [comment]
This is discussed 2 years ago [here](https://github.com/vim/vim/issues/10205). Although I would love to have this built in.

**Comment by dezza** [comment]
I agree. The unfortunate situation with [traces.vim](https://github.com/markonm/traces.vim) is also that it has serious side-effects for other potential actions involving cmdline autocmd events.

Unfortunately this won't happen unless someone puts in the time to code this behaviour - look at other code already changing a buffer on input..
